website/content/docs/runner/on-demand-runner.mdx

```diff
@@ -0,0 +1,168 @@
+and send it immediately to the new runner.
+
+The flow of information about the job will use the existing grpc runner APIs so
+all the input and output remain the same as today.
+
+Upon finishing the job, the on-demand runner will exit and the platform it runs
+on will reclaim its resources.
+
+### Launching Tasks to Execute Jobs
+
+When Waypoint goes to launch an on-demand runner task, it will generate three
+jobs:
+
+- `StartJob`
+- `TaskJob`
+- `StopJob`
+
+The start job is responsible for scheduling a resource in the platform to execute
+the assigned job. Once that resource exists, it will be a `TaskJob` that is
+responsible for actually running the assigned job. Once it's complete, the task
+plugin system will launch a stop job to end the launched platform resource.
+
+## On-Demand Runner Profiles
+
+Users can configure Waypoint server to launch on-demand runners with specific
+configurations. These configurations are set through on-demand runner profiles.
+See [runner profiles](/docs/runner/profiles) for more information.
+
+## OCI/Docker Image
+
+By default, the server will launch on-demand runners using the same OCI image
+it was configured with. The on-demand configuration will also accept a specific
+OCI image url to use instead. This allows operators the ability to prepackage
+any Waypoint plugins into the image and use them in their configuration.
+
+The official on-demand runner container image is built very similar to Waypoint
+server, with some additional packages such as `Kaniko` for doing in-container
+builds for jobs like the `Build operation`.
+
+Users can configure Waypoint server to launch on-demand runners with a custom
+built on-demand runner image. This image name must be set via the on-demand
+runner profile.
+
+## Official Waypoint Supported Task Plugin Platforms
+
+When you use the `waypoint server install` command, we automatically set up a
+default On-Demand runner profile for you to use with Waypoint Server. Like our
+server install platform support, we also currently support these platforms
+for launching on-demand runner tasks.
+
+- [Kubernetes](/plugins/kubernetes#kubernetes-task)
+- [HashiCorp/Nomad](/plugins/nomad#nomad-task)
+- [AWS/ECS](/plugins/aws-ecs#aws-ecs-task)
+- [Docker](/plugins/docker#docker-task)
+
+## Task Plugin
+
+Launching on-demand runners can be developed with the Task plugin. If there
+is a platform that you wish to support while using Waypoint, we provide a
+plugin system, similar to our build/deploy/release plugins, that lets you
+define your own configuration for launching tasks on your platform of choice.
+
+Plugins expect to fulfill the SDK interface with the following functions:
+
+- `StartTask`
+- `StopTask`
+
+These functions are what will launch and stop the spawned resource with the
+task launcher.
+
+Task plugins also can accept `TaskLauncherConfig` configuration which users can
+configure through an [on-demand runner profile](/docs/runner/profiles).
+
+## Task Introspection
```

Comment:
Similar to @krantzinator’s comment in Slack, perhaps we should ship this change alongside 0.9? Readers of the documentation might not know when to expect 0.9, and therefore when to expect this feature.